presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1937,7 +1938,7 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
                                 hivePartitionResult.getBucketHandle(),
                                 hivePartitionResult.getBucketFilter(),
                                 true,
-                                createTableLayoutString(session, tableName, hivePartitionResult.getBucketHandle(), decomposedFilter.getRemainingExpression(), domainPredicate))),
+                                createTableLayoutString(session, tableName, hivePartitionResult.getBucketHandle(), hivePartitionResult.getBucketFilter(), decomposedFilter.getRemainingExpression(), domainPredicate))),
```

Comment:
would you put this change into a separate commit?

Reply:
Done